web/pgadmin/utils/driver/psycopg3/connection.py

```
@@ -342,30 +342,58 @@ def connect(self, **kwargs):
+                        # Create/regenerate connection string (gets fresh IAM token)
+                        connection_string = manager.create_connection_string(
+                            database, user, password)
+
+                        if self.async_:
+                            autocommit = True
+                            if 'auto_commit' in kwargs:
+                                autocommit = kwargs['auto_commit']
+
+                            async def connectdbserver():
+                                return await psycopg.AsyncConnection.connect(
+                                    connection_string,
```

Ruff (0.14.14)

[warning] Function definition does not bind loop variable `connection_string`

(B023)